src/components/CellComponent.js

```diff
@@ -12,11 +12,15 @@ export default class CellComponent extends Component {
     }
 
     render() {
-        const {passable, type, terrain} = this.props;
+        const {type, terrain, fogVisibility} = this.props;
         const cellClasses = classNames(
-            'cell', passable ? 'passable' : 'impassable', 
-            {player: type === 'player', end: type === 'end'}, 
-            {mountain: terrain === Cell.Terrain.MOUNTAIN && type === Cell.Terrain.NORMAL && passable}
+            'cell', 
+            terrain !== 'wall' ? 'passable' : 'impassable', 
+            {player: type === 'player', 
+            end: type === 'end', 
+            mountain: terrain === Cell.Terrain.MOUNTAIN && type === 'normal',
+            seen: fogVisibility === 'seen', 
```

Comment:
one thing we could do is straight pass in `fogVisibility` since `classNames()` correctly handles falsy values and the resulting classname is the same as the value

in that case we may consider turning `fogVisibility` into a generic `className`, or optionally a `fogClassName` if you so desire

Reply:
🐇